pypeit/core/coadd.py

```diff
@@ -549,29 +551,40 @@ def solve_poly_ratio(wave, flux, ivar, flux_ref, ivar_ref, norder, mask = None,
     # Determine an initial guess
     ratio = robust_median_ratio(flux, ivar, flux_ref, ivar_ref, mask=mask, mask_ref=mask_ref,
                                 ref_percentile=ref_percentile, max_factor=scale_max)
+    wave_min = wave.min()
+    wave_max = wave.max()
+
     if 'poly' in model:
         guess = np.append(ratio, np.zeros(norder-1))
+        scale_fun = lambda x: x
     elif 'square' in model:
         guess = np.append(np.sqrt(ratio), np.zeros(norder-1))
+        scale_fun = np.sqrt
     elif 'exp' in model:
         guess = np.append(np.log(ratio), np.zeros(norder-1))
+        scale_fun = np.log
     else:
         msgs.error('Unrecognized model type')
 
-    wave_min = wave.min()
-    wave_max = wave.max()
 
     # Now compute median filtered versions of the spectra which we will actually operate on for the fitting. Note
     # that rejection will however work on the non-filtered spectra.
     med_width = (2.0*np.ceil(median_frac/2.0*nspec) + 1).astype(int)
     flux_med, ivar_med = median_filt_spec(flux, ivar, mask, med_width)
     flux_ref_med, ivar_ref_med = median_filt_spec(flux_ref, ivar_ref, mask_ref, med_width)
 
+    lr = scipy.stats.linregress(wave[mask], scale_fun(flux_ref_med[mask]))
```

Comment:
Please add a comment to describe what you are doing here. Also, these lines of code fail if norder = 1, which I think was actually never a use case (since we would use robust_median_ratio for that), but you should maybe put a line at the beginning causing the code to fault if border =1, and also add that to the docs. Maybe also clarify please in the docs that norder=2 is a linear fit.

Reply:
Nice catch!
It looks like in `core.coadd.scale_spec` https://github.com/pypeit/PypeIt/blob/develop/pypeit/core/coadd.py#L1275 we do use norder=1. I have a couple ideas on how to address this.

1. make `norder` intuitive, i.e. `norder=1` is a linear fit
   a. keep `scale_spec` as is (changing behavior)
   b. adjust scale spec (keep its behavior) and add a guard for `norder=1` before my linear guess code
2. keep `norder` as is, add a guard for `norder=1` before my linear guess code.

Reply:
Let's chat on Slack.